### ISSUE TITLE
no metrics collected from mongodb - uuid: null

### ISSUE THREAD

**Issue by timoweiss** [opbeat/opbeat-node#38]
I try to receive reports for a microservice which uses the module `mongodb`.
Unfortunately nothing is showing up in the `Performance Metrics` section.
 
Setting DEBUG=opbeat

```
ms-messenger-26 (err): 2016-02-19 21:18 +01:00: Fri, 19 Feb 2016 20:18:22 GMT opbeat shimming https@4.2.3 module
ms-messenger-26 (err): 2016-02-19 21:18 +01:00: Fri, 19 Feb 2016 20:18:22 GMT opbeat shimming https.Server.prototype functions: [ 'on', 'addListener' ]
ms-messenger-26 (err): 2016-02-19 21:18 +01:00: Fri, 19 Feb 2016 20:18:22 GMT opbeat shimming https.request function
ms-messenger-26 (err): 2016-02-19 21:18 +01:00: Fri, 19 Feb 2016 20:18:22 GMT opbeat shimming mongodb-core@1.2.31 module
ms-messenger-26 (err): 2016-02-19 21:18 +01:00: Fri, 19 Feb 2016 20:18:22 GMT opbeat shimming mongodb-core.Server.prototype.command
ms-messenger-26 (err): 2016-02-19 21:18 +01:00: Fri, 19 Feb 2016 20:18:22 GMT opbeat shimming mongodb-core.Server.prototype functions: [ 'insert', 'update', 'remove', 'auth' ]
ms-messenger-26 (err): 2016-02-19 21:18 +01:00: Fri, 19 Feb 2016 20:18:22 GMT opbeat shimming mongodb-core.Cursor.prototype functions: [ '_find', '_getmore' ]
ms-messenger-26 (err): 2016-02-19 21:18 +01:00: Fri, 19 Feb 2016 20:18:23 GMT opbeat shimming mongodb-core@1.2.32 module
ms-messenger-26 (err): Fri, 19 Feb 2016 20:18:23 GMT opbeat shimming mongodb-core.Server.prototype.command
ms-messenger-26 (err): Fri, 19 Feb 2016 20:18:23 GMT opbeat shimming mongodb-core.Server.prototype functions: [ 'insert', 'update', 'remove', 'auth' ]
ms-messenger-26 (err): Fri, 19 Feb 2016 20:18:23 GMT opbeat shimming mongodb-core.Cursor.prototype functions: [ '_find', '_getmore' ]
ms-messenger-26 (err): 2016-02-19 21:18 +01:00: Fri, 19 Feb 2016 20:18:23 GMT opbeat intercepted call to mongodb-core.Server.prototype.command { uuid: null, ns: 'admin.$cmd' }
ms-messenger-26 (err): 2016-02-19 21:18 +01:00: Fri, 19 Feb 2016 20:18:23 GMT opbeat intercepted call to mongodb-core.Server.prototype.command { uuid: null, ns: 'admin.$cmd' }
ms-messenger-26 (err): 2016-02-19 21:18 +01:00: Fri, 19 Feb 2016 20:18:25 GMT opbeat intercepted call to http.request { uuid: null }
ms-messenger-26 (err): 2016-02-19 21:18 +01:00: Fri, 19 Feb 2016 20:18:25 GMT opbeat intercepted call to mongodb-core.Cursor.prototype._find { uuid: null }
ms-messenger-26 (err): 2016-02-19 21:18 +01:00: Fri, 19 Feb 2016 20:18:25 GMT opbeat intercepted call to mongodb-core.Server.prototype.insert { uuid: null, ns: 'locator.conversations' }
ms-messenger-26 (err): 2016-02-19 21:18 +01:00: Fri, 19 Feb 2016 20:18:25 GMT opbeat intercepted call to mongodb-core.Server.prototype.command { uuid: null, ns: 'locator.$cmd' }
ms-messenger-26 (err): 2016-02-19 21:18 +01:00: Fri, 19 Feb 2016 20:18:25 GMT opbeat intercepted call to mongodb-core.Server.prototype.command { uuid: null, ns: 'locator.$cmd' }
ms-messenger-26 (err): 2016-02-19 21:18 +01:00: Fri, 19 Feb 2016 20:18:25 GMT opbeat intercepted call to mongodb-core.Server.prototype.insert { uuid: null, ns: 'locator.messages' }
ms-messenger-26 (err): 2016-02-19 21:18 +01:00: Fri, 19 Feb 2016 20:18:25 GMT opbeat intercepted call to mongodb-core.Server.prototype.command { uuid: null, ns: 'locator.$cmd' }

ms-messenger-26 Fri, 19 Feb 2016 20:18:55 GMT opbeat intercepted call to mongodb-core.Server.prototype.insert { uuid: null, ns: 'locator.conversations' }
ms-messenger-26 Fri, 19 Feb 2016 20:18:55 GMT opbeat intercepted call to mongodb-core.Server.prototype.command { uuid: null, ns: 'locator.$cmd' }
ms-messenger-26 Fri, 19 Feb 2016 20:18:55 GMT opbeat intercepted call to mongodb-core.Server.prototype.command { uuid: null, ns: 'locator.$cmd' }

```
I'm using `"mongodb": "^2.1.3"` and `"opbeat": "^3.5.2"`

In our API-Service which uses hapi11, everything seem to work.

**Comment by watson** [comment]
@timoweiss When the UUID of the transaction is `null` it means it could not build a new trace. Unfortunately there are several reasons why this can occur. I've published a patch to the agent (v3.5.3) which will output a little more debug-info in this particular case. Can I get you to upgrade to that version and try again. It will not fix your problem of cause, but it will allow me to see in more details what's going on :smiley:

**Comment by timoweiss** [comment]
Thanks, I now get the following:
```
ms-messenger-26 (err): 2016-02-22 13:25 +01:00: Mon, 22 Feb 2016 12:25:21 GMT opbeat agent configured correctly { node: 'v4.2.3', agent: '3.5.3', org: 'ed797675dea04db88c698adb305572c1', app: '575bf23680', instrument: true }
ms-messenger-26 (err): 2016-02-22 13:25 +01:00: Mon, 22 Feb 2016 12:25:21 GMT opbeat shimming Module._load function
ms-messenger-26 (err): 2016-02-22 13:25 +01:00: Mon, 22 Feb 2016 12:25:22 GMT opbeat shimming http@4.2.3 module
ms-messenger-26 (err): 2016-02-22 13:25 +01:00: Mon, 22 Feb 2016 12:25:22 GMT opbeat shimming http.Server.prototype functions: [ 'on', 'addListener' ]
ms-messenger-26 (err): 2016-02-22 13:25 +01:00: Mon, 22 Feb 2016 12:25:22 GMT opbeat shimming http.request function
ms-messenger-26 (err): 2016-02-22 13:25 +01:00: Mon, 22 Feb 2016 12:25:22 GMT opbeat shimming https@4.2.3 module
ms-messenger-26 (err): 2016-02-22 13:25 +01:00: Mon, 22 Feb 2016 12:25:22 GMT opbeat shimming https.Server.prototype functions: [ 'on', 'addListener' ]
ms-messenger-26 (err): 2016-02-22 13:25 +01:00: Mon, 22 Feb 2016 12:25:22 GMT opbeat shimming https.request function
ms-messenger-26 (err): 2016-02-22 13:25 +01:00: Mon, 22 Feb 2016 12:25:22 GMT opbeat shimming mongodb-core@1.2.31 module
ms-messenger-26 (err): 2016-02-22 13:25 +01:00: Mon, 22 Feb 2016 12:25:22 GMT opbeat shimming mongodb-core.Server.prototype.command
ms-messenger-26 (err): 2016-02-22 13:25 +01:00: Mon, 22 Feb 2016 12:25:22 GMT opbeat shimming mongodb-core.Server.prototype functions: [ 'insert', 'update', 'remove', 'auth' ]
ms-messenger-26 (err): 2016-02-22 13:25 +01:00: Mon, 22 Feb 2016 12:25:22 GMT opbeat shimming mongodb-core.Cursor.prototype functions: [ '_find', '_getmore' ]
ms-messenger-26 (err): 2016-02-22 13:25 +01:00: Mon, 22 Feb 2016 12:25:23 GMT opbeat shimming mongodb-core@1.2.32 module
ms-messenger-26 (err): 2016-02-22 13:25 +01:00: Mon, 22 Feb 2016 12:25:23 GMT opbeat shimming mongodb-core.Server.prototype.command
ms-messenger-26 (err): 2016-02-22 13:25 +01:00: Mon, 22 Feb 2016 12:25:23 GMT opbeat shimming mongodb-core.Server.prototype functions: [ 'insert', 'update', 'remove', 'auth' ]
ms-messenger-26 (err): 2016-02-22 13:25 +01:00: Mon, 22 Feb 2016 12:25:23 GMT opbeat shimming mongodb-core.Cursor.prototype functions: [ '_find', '_getmore' ]
ms-messenger-26 Mon, 22 Feb 2016 12:25:23 GMT opbeat no active trace found - cannot build new trace
ms-messenger-26 Mon, 22 Feb 2016 12:25:23 GMT opbeat intercepted call to mongodb-core.Server.prototype.command { uuid: null, ns: 'admin.$cmd' }
ms-messenger-26 Mon, 22 Feb 2016 12:25:23 GMT opbeat no active trace found - cannot build new trace
ms-messenger-26 Mon, 22 Feb 2016 12:25:23 GMT opbeat intercepted call to mongodb-core.Server.prototype.command { uuid: null, ns: 'admin.$cmd' }

ms-messenger-26 Mon, 22 Feb 2016 12:25:25 GMT opbeat no active trace found - cannot build new trace
ms-messenger-26 Mon, 22 Feb 2016 12:25:25 GMT opbeat intercepted call to mongodb-core.Cursor.prototype._find { uuid: null }
```

**Comment by watson** [comment]
@timoweiss thanks :smiley: I'll take a look at it

**Comment by watson** [comment]
@timoweiss This have been a tough bug to hunt down as it's related to how we instrument the Node.js event loop. I just implemented a fix that will solve this issue in a lot of the cases. It's published as v3.6.6 on npm. Please give it a spin and let me know if it fixes your particular issue :smiley:

**Comment by watson** [comment]
A lot of changes have been made to this in the latest version (v3.6.10). So this should hopefully be fixed now.

I'm closing the issue for now, but feel free to re-open if it hasn't fixed it 😃
